receiver/otlpreceiver/internal/arrow/arrow.go

```diff
@@ -135,64 +146,85 @@ func newHeaderReceiver(streamCtx context.Context, includeMetadata bool) *headerR
 // client.Info with additional key:values associated with the arrow batch.
 // This is safe to call when h is nil.
```

Comment:
It looks like we're removing the nil check on `h` here, is that okay to do?

Reply:
Right, there is no longer a nil in this setting. Fixed the comment.